server/settings_test.go

```diff
@@ -72,6 +73,188 @@ func TestSettings(t *testing.T) {
+				defer teardown(t)
+
+				// Ensure Telemetry is enabled
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableTelemetry: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+
+				res, err = serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				resg, err := serverClient.Default.Server.GetSettings(&server.GetSettingsParams{
+					Body:    server.GetSettingsDefaultBody{},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, resg.Payload.Settings.TelemetryEnabled)
+
+				// Restore the STT state to false
+				res, err = serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: false,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+			})
+
+			t.Run("DisableSTTWhileItIsDisabled", func(t *testing.T) {
+				defer teardown(t)
+
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						DisableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.False(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+			})
+			t.Run("STTEnabledState", func(t *testing.T) {
+				defer teardown(t)
+
+				_, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.SttEnabled)
+				assert.Empty(t, err)
+
+				t.Run("EnableSTTWhileItIsEnabled", func(t *testing.T) {
+					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+						Body: server.ChangeSettingsBody{
+							EnableStt: true,
+						},
+						Context: pmmapitests.Context,
+					})
+					require.NoError(t, err)
+					assert.True(t, res.Payload.Settings.SttEnabled)
+					assert.Empty(t, err)
+				})
+
+				t.Run("DisableTelemetryWhileSTTEnabled", func(t *testing.T) {
+					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+						Body: server.ChangeSettingsBody{
+							DisableTelemetry: true,
+						},
+						Context: pmmapitests.Context,
+					})
+					pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, `cannot disable telemetry while STT is enabled`)
+					assert.Empty(t, res)
+				})
+			})
+
+			t.Run("DisableSTTWhileItIsDisabled", func(t *testing.T) {
```

Comment:
looks like duplicated test

Reply:
we do not have such test before, thats basically the Current state check

Reply:
https://github.com/Percona-Lab/pmm-api-tests/pull/91/files#diff-356be822aecac50b5a0a1a1b9efe0676R200

Reply:
thanks, missed that one